components/org.wso2.carbon.identity.discovery/src/main/java/org/wso2/carbon/identity/discovery/builders/ProviderConfigBuilder.java

```diff
@@ -126,8 +125,12 @@ public OIDProviderConfigResponse buildOIDProviderConfig(OIDProviderRequest reque
         } catch (IdentityOAuth2Exception e) {
             throw new ServerConfigurationException("Unsupported signature algorithm configured.", e);
         }
-        providerConfig.setTokenEndpointAuthMethodsSupported(
-                OAuth2Util.getSupportedClientAuthenticationMethods().stream().toArray(String[]::new));
```

Comment:
the util method `getSupportedClientAuthenticationMethods` should be improved to retrieve the supported methods. 

Additionally, instead of getting these from a server config is there a way to get these values dynamically from the actually available client authentication options in the server?

Reply:
We can get the authenticators dynamically, but the issue we come across then is that we do not have a method of mapping the authenticator class with the authenticator referring name. As an example for the PrivateKeyJWT authenticator we use the referring term as 'private_key_jwt' which cannot be mapped with the authenticator. But there is a way to overcome this which is by adding a constant to each authenticator with the referring name and then use that constant to do the mapping. Shall we go ahead with that approach? WDYT?

Reply:
If not we can write a separate method from which we could do that mapping based on the class name of the authenticator. But the issue we come across then is that if a custom authenticator is added, it will be skipped as the mappings will be hardcoded in this said method.

Reply:
Shall we add a new method to the interface so that implementing authenticators should define the this reference name.

Reply:
Yeah. Thats better. I will create separate PRs to add it to the interface and to update the existing authenticators.

Reply:
This concern was addressed from the following PRs.
[wso2-extensions/identity-inbound-auth-oauthh#2188](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2188)
https://github.com/wso2-extensions/identity-oauth-addons/pull/110